MobiFlight/MobiFlightCache.cs

```diff
@@ -277,33 +277,38 @@ public bool Available()
         /// as opposed to being connected as COM port.
         /// </summary>
         /// <returns>The list of connected USB drives supported by MobiFlight.</returns>
-        public static List<MobiFlightModuleInfo> FindConnectedUsbDevices()
+        public static List<MobiFlightModuleInfo> FindConnectedUsbDevices(double WaitInMilliseconds = 500)

```

Comment:
There should be some comments in the code explaining why all this was necessary, referencing back to this bug. Future us, looking at this code, will have no idea why there's a task involved at all.